cmd/cosign/cli/verify/verify_blob_test.go

```diff
@@ -959,6 +957,91 @@ func TestVerifyBlobCmdWithBundle(t *testing.T) {
 	})
 }
 
+func TestVerifyBlobCmdInvalidRootCA(t *testing.T) {
+	keyless := newKeylessStack(t)
+	// Change the keyless stack.
+	_ = newKeylessStack(t)
+	t.Run("Invalid certificate root explicit certRef", func(t *testing.T) {
+		// Change the keyless stack.
+		_ = newKeylessStack(t)
+		identity := "hello@foo.com"
+		issuer := "issuer"
+		leafCert, _, leafPemCert, signer := keyless.genLeafCert(t, identity, issuer)
+
+		// Create blob
+		blob := "someblob"
+
+		// Sign blob with private key
+		sig, err := signer.SignMessage(bytes.NewReader([]byte(blob)))
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		// Create bundle
+		entry := genRekorEntry(t, hashedrekord.KIND, hashedrekord.New().DefaultVersion(), []byte(blob), leafPemCert, sig)
+		b := createBundle(t, sig, leafPemCert, keyless.rekorLogID, leafCert.NotBefore.Unix()+1, entry)
+		b.Bundle.SignedEntryTimestamp = keyless.rekorSignPayload(t, b.Bundle.Payload)
+		bundlePath := writeBundleFile(t, keyless.td, b, "bundle.json")
+		blobPath := writeBlobFile(t, keyless.td, blob, "blob.txt")
+		certPath := writeBlobFile(t, keyless.td, string(leafPemCert), "cert.pem")
+
+		// Verify command
+		err = VerifyBlobCmd(context.Background(),
+			options.KeyOpts{BundlePath: bundlePath},
+			certPath, /*certRef*/
+			identity, /*certEmail*/
+			issuer,   /*certOidcIssuer*/
+			"",       /*certChain*/ // Chain is fetched from TUF/SIGSTORE_ROOT_FILE
+			"",       /*sigRef*/    // Sig is fetched from bundle
+			blobPath, /*blobRef*/
+			// GitHub identity flags start
+			"", "", "", "", "",
+			// GitHub identity flags end
+			false /*enforceSCT*/)
+		if err == nil || !strings.Contains(err.Error(), "certificate signed by unknown authority") {
+			t.Fatalf("expected error with invalid root CA, got %v", err)
+		}
+	})
+	t.Run("Invalid certificate root explicit bundle", func(t *testing.T) {
```

Comment:
nit: maybe "Invalid certificate root, certificate fetched from bundle"? Took me a while to puzzle out what's going on

Reply:
"Invalid certificate root when specifying cert in bundle"